### ISSUE TITLE
Auto focus after clicking on share shows drop down list and scrolling issue

### ISSUE THREAD

**Issue by AInteriorB** [nextcloud/server#15261]
Sometimes you just would like to open the file inspector to copy an already created shared link or to display file details. By clicking on the file (or even by clicking on "details"), the auto focus on "username or e-mail" fires the drop down list with users and groups to select. This hides your created shared links. 

The behaviour is becoming worse if you start scrolling. The drop down has no fix position next to the input textfiled. Instead it starts scrolling as well.

### Steps to reproduce
1. click on a file in list (or on share, or on date, or on "details")
2. file inspector is shown

### Expected behaviour
nothing else

### Actual behaviour
1. auto focus on input filed -> drop down list appears --> is hiding essential information 
2. on scrolling the list is also scrolling

### Server configuration

**Operating system**: Debian 10
**Web server:** Apache/2.4.38
**Database:** MariaDB 10.3.14
**PHP version:** PHP 7.3.4
**Nextcloud version:** 16.0.0
**Updated from an older Nextcloud/ownCloud or fresh install:** update
**Where did you install Nextcloud from:** web updater

**Signing status:**
<details>
<summary>Signing status</summary>
No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>
Enabled:
  - accessibility: 1.2.0
  - activity: 2.9.1
  - audioplayer: 2.7.0
  - cloud_federation_api: 0.2.0
  - comments: 1.6.0
  - dav: 1.9.2
  - federatedfilesharing: 1.6.0
  - files: 1.11.0
  - files_pdfviewer: 1.5.0
  - files_rightclick: 0.13.0
  - files_sharing: 1.8.0
  - files_texteditor: 2.8.0
  - files_trashbin: 1.6.0
  - files_versions: 1.9.0
  - files_videoplayer: 1.5.0
  - gallery: 18.3.0
  - logreader: 2.1.0
  - lookup_server_connector: 1.4.0
  - nextcloud_announcements: 1.5.0
  - notifications: 2.4.1
  - oauth2: 1.4.2
  - password_policy: 1.6.0
  - privacy: 1.0.0
  - provisioning_api: 1.6.0
  - recommendations: 0.4.0
  - serverinfo: 1.6.0
  - sharebymail: 1.6.0
  - support: 1.0.0
  - survey_client: 1.4.0
  - systemtags: 1.6.0
  - theming: 1.7.0
  - twofactor_backupcodes: 1.5.0
  - updatenotification: 1.6.0
  - viewer: 1.0.0
  - workflowengine: 1.6.0
Disabled:
  - admin_audit
  - encryption
  - federation
  - files_external
  - firstrunwizard
  - user_ldap
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** yes

**Are you using an external user-backend, if yes which one:** no

**Comment by juliushaertl** [comment]
@ChristophWurst @nextcloud/designers This is something we should improve on with the new sharing suggestions. The share links should be visible even if the suggestions are presented to the user.

**Comment by ChristophWurst** [comment]
So

* Sharing dropdown opens by default. This is because the input is focussed automatically when the sharing sidebar opens <- works as designed. This is not a bug
* Sharing dropdown does break on scroll <- this is a bug with jQuery autocomplete. I hope we can replace this with a decent Vue component soonish

The latter should actually be fixed with the *unified sharing component* for Nextcloud 17. Is that what you are referring to with "new sharing suggestions"?

**Comment by jancborchardt** [comment]
Totally agree that this is an issue, brought as a combination of the autofocus of the "Share" field (that has been there forever) in combination with the new automatic showing of suggestions (which is new and nice).

As a simple fix, the share input should not be focused automatically anymore on opening the share tab / clicking " Share" in the files list. The autofocus was done at a time when the input field was the only way to interact in that tab – but now we have links, projects, options for the sharees etc.

What do you think?

Regarding the scrolling bug I agree with @ChristophWurst’s take.

**Comment by AInteriorB** [comment]
Exactly, @jancborchardt ! I use the "share link" feature much more often than the "share to a user" feature. Thus, I always need an extra click to get rid of the suggestion drop down list.

A solution to have both functions (auto focus and suggestion list) would be to show the drop down list only after you start typing.

**Comment by jancborchardt** [comment]
@skjnldsv @ChristophWurst could you point me in the rough direction where this would need to be changed? Then I can do it. I searched around in `core/js/sharedialogview.js` and modified some stuff, but no effect.

**Comment by jancborchardt** [comment]
Fix at https://github.com/nextcloud/server/pull/15960 – @AInteriorB could you open a separate issue about the scrolling problem? That way we keep discussion focused. :)
